### DISCUSSION TITLE
What does asterisk (*) in a type mean?

### DISCUSSION THREAD

**Discussion by decorator-factory** [microsoft/pylance-release#1707] [General] [answered]
```py
from typing import Generic, TypeVar, Union

T = TypeVar("T")

class _Unset():
    pass
_UNSET = _Unset()


class Foo(Generic[T]):
    def __init__(self):
        self._result: Union[T, _Unset] = _UNSET

    def is_set(self):
        return self._result is not _UNSET
```
When I hover over `is_set` in VSCode with Pylance, I see this:
![(method) is_set: (self: Foo[T@Foo]) -> (bool* | bool)](https://user-images.githubusercontent.com/42166884/130348741-685d0fb4-4931-4f3b-8e2c-7a80e8ebbe9f.png)

What does `bool*` mean?

**Comment by erictraut** [comment, marked as answer]
Please refer to [this documentation](https://github.com/microsoft/pyright/blob/main/docs/type-concepts.md#constrained-type-variables-and-conditional-types) for details.

**Comment by studyingegret** [comment]
https://github.com/microsoft/pyright/blob/046eab4a8dd8344ae614f9214f0871db64085163/docs/type-concepts.md#constrained-type-variables-and-conditional-types
The latest version of the document linked by @<span></span>erictraut removed the content about starred types.

**Reply by erictraut** [reply]
This information was split out into an "advanced type concepts" page about three years ago. Here is an updated [link](https://github.com/microsoft/pyright/blob/main/docs/type-concepts-advanced.md#conditional-types-and-type-variables).
